romancal/skycell/skymap.py

```diff
@@ -0,0 +1,588 @@
+        instance = cls(index=None)
+        instance.__data = data
+        return instance
+
+    @classmethod
+    def from_center_and_coordinates(
+        cls,
+        skytile_center: tuple[float, float],
+        skycell_coordinates: tuple[float, float],
+    ) -> "SkyCell":
+        return cls.from_name(
+            f"r{round(skytile_center[0]):03}d{'p' if skytile_center[1] >= 0 else 'm'}{round(skytile_center[1]):02}x{'p' if skycell_coordinates[1] >= 0 else 'm'}{skycell_coordinates[1]:02}y{'p' if skycell_coordinates[1] >= 0 else 'm'}{skycell_coordinates[1]:02}"
+        )
+
+    @classmethod
+    def from_modellibrary(cls, library: ModelLibrary) -> "SkyCell":
+        if (
+            "skycell_wcs_info" in library.asn
+            and library.asn["skycell_wcs_info"] != "none"
+        ):
+            skycell_name = library.asn["skycell_wcs_info"]["name"]
+        elif "target" in library.asn:
+            # check to see if the product name contains a skycell name & if true get the skycell record
+            skycell_name = library.asn["target"]
+        else:
+            raise ValueError(
+                "cannot extract skycell information from modellibrary association with neither WCS nor target info"
+            )
+
+        return SkyCell(skycell_name)
+
+    @property
+    def index(self) -> int | None:
+        return self.__index
+
+    @property
+    def data(self) -> np.void:
+        return self.__data
+
+    @property
+    def name(self) -> str:
+        return self.data[0]
+
+    @property
+    def radec_center(self) -> tuple[float, float]:
+        return self.data["ra_center"], self.data["dec_center"]
+
+    @property
+    def orientation(self) -> float:
+        return self.data["orientat"]
+
+    @property
+    def xy_tangent(self) -> tuple[float, float]:
+        return self.data["x_tangent"], self.data["y_tangent"]
+
+    @property
+    def radec_corners(
+        self,
+    ) -> NDArray[float]:
+        return np.array(
+            (
+                (self.data["ra_corn1"], self.data["dec_corn1"]),
+                (self.data["ra_corn2"], self.data["dec_corn2"]),
+                (self.data["ra_corn3"], self.data["dec_corn3"]),
+                (self.data["ra_corn4"], self.data["dec_corn4"]),
+            )
+        )
+
+    @property
+    def polygon(self) -> sgp.SingleSphericalPolygon:
+        # convert all radec points to vectors
+        corner_vectorpoints = image_coords_to_vec(self.radec_corners)
+        center_vectorpoint = image_coords_to_vec([self.radec_center])
+
+        # construct polygon from corner points and center point
+        return sgp.SingleSphericalPolygon(
+            points=sgv.normalize_vector(corner_vectorpoints),
+            inside=sgv.normalize_vector(center_vectorpoint),
+        )
+
+    @cached_property
+    def skytile(self) -> "SkyTile":
+        if self.index is None:
+            raise ValueError("no index provided")
+        return SkyTile.from_skycell_index(self.index)
```

Comment:
`SkyTile.from_skycell_index(0)` fails for me with 
`'sky cell index 0 not found in any projection regions' ` Am I running it incorrectly?